includes/class-wsuwp-vals-custom-roles.php

```diff
@@ -26,5 +47,588 @@ public static function get_instance() {
+	function update_center_count( $terms, $taxonomy ) {
+		global $wpdb;
+
+		foreach ( (array) $terms as $term ) {
+
+			$count = $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(*) FROM $wpdb->term_relationships WHERE term_taxonomy_id = %d", $term ) );
+
+			$wpdb->update( $wpdb->term_taxonomy, compact( 'count' ), array( 'term_taxonomy_id' => $term ) );
+
+			do_action( 'edit_term_taxonomy', $term, $taxonomy );
+			do_action( 'edited_term_taxonomy', $term, $taxonomy );
+		}
+	}
+
+	/**
+	 * Return the certification with a wrapper indicating its status.
+	 *
+	 * @since 0.0.1
+	 *
+	 * @param string $certification The user's certification date.
+	 *
+	 * @return string $certification The user's certification date.
+	 */
+	public function certification_status( $certification ) {
+		$certification_date = new DateTime( $certification );
+		$today = new DateTime( 'now' );
+		$difference = $certification_date->diff( $today );
+
+		if ( 5 < (int) $difference->y ) {
+			// User's certification has expired.
+			return '<span style="color:#c60c30;">' . esc_html( $certification ) . '</span>';
+		} elseif ( 54 < (int) 12 * $difference->y + $difference->m ) {
+			// User's certification expires in 6 (or fewer) months.
+			return '<span style="color:#f6861f;">' . esc_html( $certification ) . '</span>';
+		} else {
+			return esc_html( $certification );
+		}
+	}
+
+	/**
+	 * Add a 'VALS Data' section to the edit user/profile page.
+	 *
+	 * @since 0.0.1
+	 *
+	 * @param object $user The user object currently being edited.
+	 */
+	public function extend_user_profile( $user ) {
+		// Bail if not viewing a profile for a user with a custom VALS role.
+		if ( ! array_intersect( $this->roles, (array) $user->roles ) ) {
+			return;
+		}
+
+		wp_nonce_field( 'save-vals-user-data', '_vals_user_nonce' );
+
+		?>
+		<tr class="vals-data">
+			<td colspan="2">
+				<h2>VALS Data</h2>
+			</td>
+		</tr>
+		<?php
+		$taxonomy = get_taxonomy( $this->taxonomy_slug );
+		$terms = get_terms( $this->taxonomy_slug, array( 'hide_empty' => false ) );
+		?>
+		<tr class="vals-data">
+			<th><label for="<?php echo esc_attr( $this->taxonomy_slug ); ?>">Center</label></th>
+			<td id="<?php echo esc_attr( $this->taxonomy_slug ); ?>"><?php
+			if ( current_user_can( $taxonomy->cap->assign_terms ) && ! empty( $terms ) ) {
+				foreach ( $terms as $term ) {
+					?>
+					<input type="radio"
+						   name="<?php echo esc_attr( $this->taxonomy_slug ); ?>"
+						   id="<?php echo esc_attr( $this->taxonomy_slug . '-' . $term->slug ); ?>"
+						   value="<?php echo esc_attr( $term->slug ); ?>"
+							<?php checked( true, is_object_in_term( $user->ID, $this->taxonomy_slug, $term ) ); ?> />
+					<label for="<?php echo esc_attr( $this->taxonomy_slug . '-' . $term->slug ); ?>"><?php echo esc_html( $term->name ); ?></label><br />
+					<?php
+				}
+			} else {
+				$center = wp_get_object_terms( $user->ID, $this->taxonomy_slug );
+				?><p><?php echo esc_html( $center[0]->name ); ?></p><?php
```

Comment:
This will only matter in specific scenarios, like me not having any test centers added yet... but - If no terms have been created, then the output of the name will fail here because `$center` is an empty array.